app/src/main/java/com/sdp13epfl2021/projmag/activities/CommentsActivity.kt

```diff
@@ -1,24 +1,87 @@
 import com.sdp13epfl2021.projmag.adapter.MessageListAdapter
+import com.sdp13epfl2021.projmag.database.interfaces.CommentsDatabase
+import com.sdp13epfl2021.projmag.database.interfaces.ProjectId
+import com.sdp13epfl2021.projmag.database.interfaces.UserdataDatabase
 import com.sdp13epfl2021.projmag.model.*
+import dagger.hilt.android.AndroidEntryPoint
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Named
 
+@AndroidEntryPoint
 class CommentsActivity : AppCompatActivity() {
     private lateinit var recyclerView: RecyclerView
     private lateinit var messageAdapter: MessageListAdapter
+    private lateinit var comments : List<Message>
+
+
+
+    @Inject
+    lateinit var  commentsDB : CommentsDatabase
+
+    @Inject
+    lateinit var userDB: UserdataDatabase
+
+    @Inject
+    @Named("currentUserId")
+    lateinit var userId: String
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_comments)
-
-        val comments = MessagesDatasource().loadMessages()
+        //get the users profileId
+        val profileId : String = userId
+        // send a message
+        // get the project id
+        val projectId = intent.extras?.getString("projectId")
+        setUpSendButton(profileId, projectId)
         val recyclerView = findViewById<RecyclerView>(R.id.recycler_view_comments)
-        recyclerView.adapter = MessageListAdapter(this, comments)
-        recyclerView.setHasFixedSize(false)
+        commentsDB.getCommentsOfProject(projectId!!, {
+            this.comments = it
+            recyclerView.adapter = MessageListAdapter(commentsDB,this, comments, userDB)
+            recyclerView.setHasFixedSize(false)
+            recyclerView.scrollToPosition(this.comments.size - 1)
+        }, {})
+
+        commentsDB.addListener(projectId) { _: ProjectId, _: List<Message> ->
+            this.runOnUiThread {
+                commentsDB.getCommentsOfProject(projectId, {
+                    this.comments = it
+                    recyclerView.adapter = MessageListAdapter(commentsDB,this, comments, userDB)
+                    recyclerView.setHasFixedSize(false)
+                    recyclerView.scrollToPosition(this.comments.size - 1)
+                }, {})
+            }
+        }
+    }
+
+    private fun setUpSendButton(profileId : String?, projectId : String?) {
+        if(profileId != null) {
+            val sendCommentButton = findViewById<ImageButton>(R.id.comments_send_button)
+            sendCommentButton.setOnClickListener {
+                val editTextView = findViewById<EditText>(R.id.comments_edit_text)
+                val messageText = editTextView.text.toString()
+                when (val message = Message.build(messageText, profileId, Date().time) ) {
+                    is Success ->
+                    {
+                        commentsDB.addCommentToProjectComments(message.value, (projectId ?: ""),{}, {})
```

Comment:
If you finish the activity if projetctId is null, this will avoid problem if null.
```suggestion
                        commentsDB.addCommentToProjectComments(message.value, projectId, {}, {})
```